subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -113,7 +114,12 @@ public synchronized void setDefaultExcludesFromSettings(String[] excludesFromSet
         }
     }
 
-    private void updateDefaultExcludeSpecCache(String[] defaultExcludes) {
+    @VisibleForTesting
+    public synchronized List<String> getDefaultExcludesFromSettings() {
+        return ImmutableList.copyOf(previousDefaultExcludes);
+    }
+
+    protected void updateDefaultExcludeSpecCache(String[] defaultExcludes) {
```

Comment:
This method can remain `private`, right?

Reply:
Yes we can!